locksmith/src/operations/receiptOperations.ts

```diff
@@ -1,29 +1,84 @@
 import { SubgraphService } from '@unlock-protocol/unlock-js'
 import { Receipt, ReceiptBase } from '../models'
+import { getMetadataByTokenId } from './userMetadataOperations'
+import { lowercaseObjectKeys } from '../utils/object'
 
 interface ReceiptDetailsProps {
   lockAddress: string
   network: number
   hash: string
+  tokenId?: string
 }
 
-export const getReceiptDetails = async ({
+/** Get purchaser details from metadata when receipt purchaser is not present */
+const getPurchaserDetails = async ({
   lockAddress,
   network,
   hash,
-}: ReceiptDetailsProps): Promise<{
-  supplier: ReceiptBase | null
-  purchaser: Receipt | null
-  receipt: any
-}> => {
-  const purchaser = await Receipt.findOne({
+  tokenId,
+}: ReceiptDetailsProps): Promise<Partial<
+  Receipt & { email?: string }
+> | null> => {
+  const receiptPurchaser = await Receipt.findOne({
     where: {
       network,
       lockAddress,
       hash,
     },
   })
 
+  let metadata: any
+  if (tokenId) {
+    metadata = await getMetadataByTokenId({
+      tokenId,
+      network,
+      lockAddress,
+      includeProtected: true,
+    })
+  }
+
+  const data = lowercaseObjectKeys(metadata?.userMetadata?.protected ?? {})
```

Comment:
```suggestion
  const data = lowercaseObjectKeys(metadata?.userMetadata?.protected ?? {})
```
move `{}` inside the function